Sources/Workspace/Workspace.swift

```diff
@@ -45,10 +45,10 @@ public protocol WorkspaceDelegate: class {
     func didLoadManifest(packagePath: AbsolutePath, url: String, version: Version?, packageKind: PackageReference.Kind, manifest: Manifest?, diagnostics: [Diagnostic])
 
     /// The workspace has started fetching this repository.
-    func fetchingWillBegin(repository: String)
+    func fetchingWillBegin(repository: String, fetchDetails: RepositoryManager.FetchDetails?)
```

Comment:
same here

Reply:
Yes, the general approach I'd recommend (to avoid breaking clients) when adding new delegate callbacks is to have the code call the new method, and then provide a default implementation of the new method that calls the old one.

Something like:
```
    public func fetchingWillBegin(repository url: String, fetchDetails: RepositoryManager.FetchDetails?) {
        self.fetchingWillBegin(repository: url)
    }
```

The clients can update at their leisure.  Marking the old callback as deprecated would be fine, I think.

Reply:
Once there are semantic versions of libSwiftPM this won't be an issue (we'd just bump the semantic version when making breaking changes), but while all the clients still have to have branch dependencies, it's difficult to stage the adoption without trying to coordinate commits across repositories.